### PR TITLE
MVP-100-Breadcrumbs

### PR DESCRIPTION
**Pull request by SDToews**
## 🎯 Summary

  
[MVP-100](https://citz-imb.atlassian.net/browse/MVP-100)

added breadcrumb trail to all pages

## 🔰 Checklist

- [X] I have read and agree with the following checklist and am following the guidelines in our [Code of Conduct](/CODE_OF_CONDUCT.md) document.

> - I have performed a self-review of my code.
> - I have commented my code, particularly in hard-to-understand areas.
> - I have made corresponding changes to the documentation where required.
> - I have tested my changes to the best of my ability.
> - I have consulted with the team if introducing a new dependency.
> - My changes generate no new warnings.

### CODE REVIEW

File: frontend/src/components/Breadcrumbs/Breadcrumbs.tsx [hunk 1]
```diff
@@ -0,0 +1,26 @@
+import { makeCrumb } from './makeCrumb';
+import { BreadCrumbProps, BreadCrumbStyles, CrumbProperty } from './types';
+import { styleMapper } from './styleMapper';
+import { useDynamicStyles } from 'hooks';
+import { Crumb } from './Crumb';
+
+export const Breadcrumbs = (props: BreadCrumbProps) => {
+  const { pathname } = props;
```
Comment (BradyMitch) [inline frontend/src/components/Breadcrumbs/Breadcrumbs.tsx:8]: pathname should probably just be defined here instead of passed as props.

**Comment by BradyMitch** [comment]
Few things:

1. When I navigate to a new page the breadcrumbs dont update until I refresh.
2. We shouldn't be displaying breadcrumbs on the home page because there is nowhere to navigate to.
3. We should add aliases so we can say `sso-react` is displayed as `SSO React` in the breadcrumbs.
4. I think we should follow the designs in the mockup or at least something similar to the gov.bc.ca styles. Especially by showing clickable links as an underlined blue text (for this you can just use the `Button` component from the `design-system` package with props `variant="link"`)

File: frontend/src/components/Breadcrumbs/Breadcrumbs.tsx (hunk 1 above)
Comment (SDToews) [reply frontend/src/components/Breadcrumbs/Breadcrumbs.tsx:8]: I passed as props to decouple the component from any dependencies on specific package libraries

File: .gitignore [hunk 2]
```diff
@@ -2,4 +2,5 @@ node_modules
 package-lock.json
 build
 .env
+/frontend/packages
```
Comment (BradyMitch) [inline .gitignore:5]: This is no longer needed with the use of the volume mount

Comment (SDToews) [reply .gitignore:5]: removed

File: frontend/jest.config.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+import type { Config } from 'jest';
+import { defaults } from 'jest-config';
+
+const config: Config = {
+  ...defaults,
+  moduleFileExtensions: ['ts', 'tsx', 'js', 'mjs', 'cjs', 'jsx', 'mts', 'cts', 'json', 'node'],
+  moduleNameMapper: {
+    '\\.(css|less|scss|sass)$': 'identity-obj-proxy',
+  },
+  modulePaths: ['<rootDir>/src/'],
```
Comment (BradyMitch) [inline frontend/jest.config.ts:10]:
```suggestion
  modulePaths: ['<rootDir>/src/'],
  coveragePathIgnorePatterns: ['index.ts'],
```

File: frontend/__tests__/unit/Breadcrumbs/Breadcrumbs.test.tsx [hunk 4]
```diff
@@ -0,0 +1,28 @@
+import { Breadcrumbs } from 'components';
+import { render } from '@testing-library/react';
+
+describe('Breadcrumbs', () => {
+  describe('when pathname is "/"', () => {
+    it('should render a single link to the home page', () => {
+      const element = render(<Breadcrumbs pathname="/" />);
```
Comment (BradyMitch) [inline frontend/__tests__/unit/Breadcrumbs/Breadcrumbs.test.tsx:7]: TypeError: Property 'labels' is missing in type '{ pathname: string; }' but required in type 'BreadCrumbProps'.

File: frontend/package.json [hunk 5]
```diff
@@ -17,16 +19,30 @@
...
+    "@testing-library/jest-dom": "6.4.2",
+    "@testing-library/react": "14.2.2",
+    "@testing-library/user-event": "14.5.2",
+    "@types/jest": "29.5.12",
     "@types/node": "20.11.30",
     "@types/react": "18.2.71",
     "@types/react-dom": "18.2.22",
     "@typescript-eslint/eslint-plugin": "7.4.0",
     "@typescript-eslint/parser": "7.4.0",
+    "babel-jest": "29.7.0",
```
Comment (BradyMitch) [inline frontend/package.json:34]:
```suggestion
    "@jest/globals": "29.7.0",
    "@testing-library/jest-dom": "6.4.2",
    "@testing-library/react": "14.2.2",
    "@testing-library/user-event": "14.5.2",
    "@types/jest": "29.5.12",
    "@types/node": "20.11.30",
    "@types/react": "18.2.71",
    "@types/react-dom": "18.2.22",
    "@typescript-eslint/eslint-plugin": "7.4.0",
    "@typescript-eslint/parser": "7.4.0",
```
Remove as they are not used

File: frontend/src/components/AuthButtons/AuthButtons.tsx [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+  if (isAuthenticated)
+    return (
+      <Button variant="secondary" onPress={() => logout()}>
+        LOGOUT
+      </Button>
+    );
+
+  return (
+    <Button variant="secondary" onPress={() => login()}>
```
Comment (BradyMitch) [inline frontend/src/components/AuthButtons/AuthButtons.tsx:15]:
```suggestion
    <Button variant="secondary" onPress={() => login({ idpHint: 'idir' })}>
```

**Comment by BradyMitch** [comment]
I like what you've done with Router and the BreadCrumbs work well and look nice, but I'm gonna have to ask you to redo this pr as smaller prs with some changes noted below.

- `RootLayout` should be under `components/RootLayout`
- type `CrumbProps` of `components/BreadCrumbs/Crumb` should be in `types.ts`
- `.d.ts` files are for declaration files where the declare keyword is used, you should just use `types.ts` instead of `types.d.ts`
- `packagedata` should probably go under a new folder `constants/packages` or something that better describes what it is and follows naming convention for filenames.
- props should not be passed to `AuthButtons`, they should be defined within the component, and rename to `AuthButton` because it returns a single button. I get that this makes the component more reusable for others using different authentication than sso but this adds complexity and we're not building this for others.
- I dont like the use of `src/types` and `STYLE_DEFAULTS` as each component is very different and this makes the components less self-contained as they rely on the `STYLE_DEFAULTS` from this other folder.

**Comment by SDToews** [comment]
Breaking into smaller pull requests

**Pull request closed without merging**